pkg/analysis/helpers/markers/registry.go

```diff
@@ -0,0 +1,83 @@
+package markers
+
+import (
+	"strings"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Registry is a thread-safe set of known marker identifiers.
+type Registry interface {
+	// Register adds the provided identifiers to the Registry.
+	Register(ids ...string)
+
+	// Match performs a greedy match to determine if an input
+	// marker string matches a known marker identifier. It returns
+	// the matched identifier and a boolean representing if a match was
+	// found. If no match is found, the returned identifier will be an
+	// empty string.
+	Match(in string) (string, bool)
+}
+
+var defaultRegistry = NewRegistry() //nolint:gochecknoglobals
```

Comment:
Shame it still needs the noglobals, but since this is private now, I think this is safe, thanks